pmm-app/src/pmm-qan-app-panel/QueryAnalytics/StateContext.tsx

```diff
@@ -0,0 +1,279 @@
+import React, { useEffect, useState } from 'react';
+import { ParseQueryParamDate } from '../../react-plugins-deps/components/helpers/time-parameters-parser';
+import { getDataSourceSrv } from '@grafana/runtime';
+import _ from 'lodash';
+
+const initialState = {} as any;
+
+export const StateContext = React.createContext(initialState);
+
+const DEFAULT_COLUMNS = ['load', 'num_queries', 'query_time'];
+const FILTERS_NAMES = [
+  'environment',
+  'cluster',
+  'replication_set',
+  'database',
+  'schema',
+  'node_name',
+  'service_name',
+  'client_host',
+  'username',
+  'service_type',
+  'node_type',
+  'city',
+  'az',
+];
+
+class ContextActions {
```

Comment:
We probably don't need a class here, since there is no intent to create instances off of it. A much better approach is to use a simple object.